### PR TITLE
libb64: Add version 2.0.0.1 (a new recipe)

### PR DESCRIPTION
**Pull request by KGrzeg**
Specify library name and version:  **libb64/2.0.0.1**

I am submitting that tiny lib as a part of the modernization process of the [OpenHome](https://github.com/openhome) stack, to make it more dev friendly. At first I would like to add the [ohNet](https://github.com/openhome/ohNet) and libb64 is his direct dependency.

As it is my first contribution to conan center index, any feedback would be very greatly appreciated :)

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/README.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by KGrzeg** [comment]
Sorry, I need some help with CI issues. I can not reproduce the error in my local environment. Tested with CLang on Linux, and on two Mac devices, everything looks fine.

**Comment by DoomHammer** [comment]
Is it possible to re-run the checks? It seems to build on my Mac machine.

**Comment by KGrzeg** [comment]
I have caused a rebuild. It looks like it doesn't interrupt immediately. I have no access to jfrog building service tho (403 error), so I can not verify it.

File: recipes/libb64/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.15)
+
+project(test_package CXX)
+
+find_package(libb64 REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} PRIVATE libb64::libb64)
```
Comment (DoomHammer) [inline recipes/libb64/all/test_package/CMakeLists.txt:8]: Shouldn't this be `PUBLIC`?

Comment (KGrzeg) [reply recipes/libb64/all/test_package/CMakeLists.txt:8]: It works as PRIVATE, so I don't think.

File: recipes/libb64/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            pattern="*.h",
+            dst=os.path.join(self.package_folder, "include"),
+            src=os.path.join(self.source_folder, "include"),
+        )
+
+        cmake = CMake(self)
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = ["libb64"]
```
Comment (DoomHammer) [inline recipes/libb64/all/conanfile.py:60]:
```suggestion
        self.cpp_info.libs = ["b64"]
```

File: recipes/libb64/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.15)
+
+project(test_package CXX)
+
+find_package(libb64 REQUIRED CONFIG)
```
Comment (DoomHammer) [inline recipes/libb64/all/test_package/CMakeLists.txt:5]:
```suggestion
find_package(b64 REQUIRED CONFIG)
```

File: recipes/libb64/all/test_package/CMakeLists.txt (hunk 1 above)
Comment (DoomHammer) [inline recipes/libb64/all/test_package/CMakeLists.txt:8]:
```suggestion
target_link_libraries(${PROJECT_NAME} PRIVATE b64::b64)
```

File: recipes/b64/all/conandata.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "1.2.1":
+    url: "https://github.com/KGrzeg/libb64-conan/archive/refs/tags/1.2.1_1.tar.gz"
```
Comment (uilianries) [inline recipes/b64/all/conandata.yml:3]: Please, use the original source from Source Forge https://sourceforge.net/projects/libb64/files/.

I see the upstream uses Makefile and VIsual Studio solution, which can be built with `Autotools` and `MSBuild` genereators/helpers.

The problem adding CMake is the maintenance, we avoid adding extra build helper files.

Comment (KGrzeg) [reply recipes/b64/all/conandata.yml]: @uilianries I have followed your advice and update my conan.py. The migration to autotools went smooth.
But there is a problem on Windows. Original sources contain visual studio solution, but the solution is not intended to build a library, but a CLI executable.
Is there any way to generate a fresh solution with conan's tools or I need to change the VS files on the fly with replace_in_file utility? (For now I have no idea what to change to obtain a desired effect though)

Comment (DoomHammer) [reply recipes/b64/all/conandata.yml]: @uilianries is it good now?

**Comment by DoomHammer** [comment]
@uilianries , can you check this, please?

**Comment by uilianries** [comment]
@KGrzeg The project from source forge stopped in 2010 and besides all bugs, no new release/interaction has been made since then. However, it seems like the project https://github.com/libb64/libb64 is its natural replacements, as mentioned in the README file:

> History
> 
> This library is based on libb64 project on sourceforge https://sourceforge.net/projects/libb64/
> As development there seems to have stopped in 2010 we forked the source to GitHub and included the patches included in https://sourceforge.net/p/libb64/bugs/
> GitHub project home can be found at https://github.com/libb64/libb64

So, I suggest moving to 2.0.0.1 version instead.

**Comment by uilianries** [comment]
@KGrzeg Please, consider https://github.com/KGrzeg/conan-center-index/pull/2 for Windows fixing

**Comment by uilianries** [comment]
@KGrzeg please, consider https://github.com/KGrzeg/conan-center-index/pull/3

**Review by uilianries — APPROVED** [review]

**Review by DoomHammer — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**